### ISSUE TITLE
Broken for Play 2.2.3

### ISSUE THREAD

**Issue by mslinn** [jamesward/play-framework-chrome-tools#18]
I have tested v0.10.0 on Windows, Mac and Linux and it no longer works with Play 2.2.3 apps. I suspect the recent update for Play 2.3.x is the reason. We cannot update Play to 2.3.x so this extension is no longer available to us :(

Is there a way to run an old version of this plugin?

**Comment by jamesward** [comment]
I think we need to fix whatever breaking changed was introduced.  cc @mariussoutier

**Comment by mariussoutier** [comment]
@mslinn What version of sbt are you using? auto-plugins require 0.13.5.

Does version 0.0.9 work for you?

**Comment by mslinn** [comment]
SBT has nothing to do with this problem.
I don't know the version that I used before that worked. How can I get old versions and install them?

**Comment by mariussoutier** [comment]
Ok then what version of the SBT plugin are you using? The chrome extension should be backwards compatible, i.e. default to port 9000 when an older version doesn't send the port.

Do you see any error message?

**Comment by mslinn** [comment]
I am not using the SBT plugin, just the Chrome Play Framework Tools plugin. There is no error message in Chrome.

**Comment by mariussoutier** [comment]
Wait, what do you use then? The chrome extension is expecting WebSocket messages sent by `play-auto-refresh`.

**Comment by mslinn** [comment]
I see I was not clear previously. I am reporting a problem with Play Framework Tools for Chrome. It is a plugin for the Chrome browser. It does not require any SBT plugins. It talks to IntelliJ IDEA. I believe the error message parsing was changed to accommodate Play 2.3.x; that broke the Chrome plugin for older versions of Play.

**Comment by jamesward** [comment]
This Chrome plugin has a few different pieces of functionality.  So the one that broke is the source link that opens the file in IntelliJ?

**Comment by mslinn** [comment]
Yes. The error message displayed in the browser no longer has a link

**Comment by jamesward** [comment]
Looks like this is what broke this for Play 2.2.x projects: https://github.com/jamesward/play-framework-chrome-tools/commit/a2dab7492be58e1341e5d28f4b8bc6b74c141f6f

I'm not sure the right way to deal with this...  Maybe try multiple styles of matching?

**Comment by mslinn** [comment]
That seems reasonable. If a regex match does not return enough groups, try matching the other regex

**Comment by mariussoutier** [comment]
Ah, funny, so you're just using that small part :) Indeed, the text that points to the error changed between 2.2 and 2.3. I was planning on updating the RegEx anyway because I had a couple of issues with it. I'll be on vacation starting tomorrow so can't fix it this week but the week after.

**Comment by jamesward** [comment]
Hey @mariussoutier - you back?  Mind giving this a shot?

**Comment by mariussoutier** [comment]
Yes sure, will start working on it tonight or tomorrow.

**Comment by jamesward** [comment]
Thanks again @mariussoutier.  Version 0.0.11 is out with your fix.  @mslinn Can you test it?  Thanks guys!

**Comment by mariussoutier** [comment]
Is there a way to force-update a Chrome Extension? I'm still on the 0.0.10.

**Comment by mariussoutier** [comment]
Got it, in developer mode, the button `Update extension now` appears.

**Comment by mariussoutier** [comment]
Yep even seems to work in an old 2.0.4 project.

**Comment by jamesward** [comment]
Awesome!  Thanks!

**Comment by mariussoutier** [comment]
But would like to have confirmation from somebody else :)

**Comment by mslinn** [comment]
No improvement using Play 2.2.4 and Play Framework Tools 0.0.11. Tested on XUbuntu and Windows 7 Pro 64 bit.

**Comment by jamesward** [comment]
So the link is not being created?

**Comment by mslinn** [comment]
The link, she is not there. I am without link. Alas, for the lack of a link a kingdom was lost. Perhaps I should write a Letter about the Link to Art. But I digress... :)

**Comment by mariussoutier** [comment]
Just to make sure, you updated to 0.0.11 and you have set an editor link in the options?

**Comment by mslinn** [comment]
Aha, an undocumented behavior which will be a problem for users when the plugin silently updates: the editor link in the options is forgotten. Once I set it the plugin worked. Thanks for fixing this, looks like you might want to do something about preserving the setting across updates.

**Comment by mariussoutier** [comment]
I'm afraid there's nothing we can do about that...

**Comment by mslinn** [comment]
You could mention it in README.md in big bold letters

**Comment by mariussoutier** [comment]
Yeah, but it's strange, my settings were not deleted.

**Comment by mslinn** [comment]
Just my luck. Somehow the worst edge cases always happen to me :(

**Comment by jamesward** [comment]
Sorry @mslinn this was such a hassle!  But really glad we got it working again.  Thanks again @mariussoutier!

**Comment by mslinn** [comment]
One more data point proving that F/OSS works :)

**Comment by mariussoutier** [comment]
My pleasure, I hope you don't mind if I port (read: copy) this to a Safari plug-in?

**Comment by jamesward** [comment]
Go for it!

**Comment by mslinn** [comment]
I was just viewing the Chrome JS console, and I now see this:

```
Error in event handler for (unknown): Cannot read property 'textContent' of undefined
Stack trace: TypeError: Cannot read property 'textContent' of undefined
    at tryPlay23 (chrome-extension://dchhggpgbommpcjpogaploblnpldbmen/play-framework-chrome-tools.js:60:68)
    at chrome-extension://dchhggpgbommpcjpogaploblnpldbmen/play-framework-chrome-tools.js:94:10
    at messageListener (extensions::messaging:340:9)
    at Function.target.(anonymous function) (extensions::SafeBuiltins:19:14)
    at EventImpl.dispatchToListener (extensions::event_bindings:397:22)
    at Function.target.(anonymous function) (extensions::SafeBuiltins:19:14)
    at Event.publicClass.(anonymous function) [as dispatchToListener] (extensions::utils:89:26)
    at EventImpl.dispatch_ (extensions::event_bindings:379:35)
    at EventImpl.dispatch (extensions::event_bindings:403:17)
    at Function.target.(anonymous function) (extensions::SafeBuiltins:19:14) 
```

Chrome Tools works, but it would be best not to see this stack trace

**Comment by mariussoutier** [comment]
It's already fixed, will be gone with the next release.

**Comment by jamesward** [comment]
I've just published `0.0.12` which should have this fix.  It might take some time for Google to propagate the change.
